drivers/virtio/net/virtio_net.c

```diff
@@ -983,10 +985,34 @@ static int virtio_netdev_feature_negotiate(struct uk_netdev *n)
 	if (VIRTIO_FEATURE_HAS(host_features, VIRTIO_F_EVENT_IDX))
 		VIRTIO_FEATURE_SET(drv_features, VIRTIO_F_EVENT_IDX);
 
+	/* Store the preliminary result in the features field.
+	 * virtio_netdev_feature_negotiate will take care of sending the result
+	 * to the host depending on how the uknetdev client configured the
+	 * device.
+	 */
+	vndev->vdev->features = drv_features;
+
+	return 0;
+err_negotiate_feature:
+	virtio_dev_status_update(vndev->vdev, VIRTIO_CONFIG_STATUS_FAIL);
```

Comment:
Why do we not check for this function's return value? :thinking: 
Actually, it seems that the whole file is doing this???? Am I missing something?

Reply:
Given the documentation, it either returns success or ENOTSUP. In the case of ENOTSUP, we wouldn't do anything. So we would ignore the return code anway.

Reply:
Sigh... the state of these virtio drivers man...

Reply:
That's not really one of the bad parts. Not supporting the status update is not a problem, so ignoring having no support is fine.

Reply:
Yes, agreed. I meant our virtio drivers in general feel a bit all over the place and there are other functions called whose returned values are not consistently checked for, like `uk_sglist_append` or `virtio_netdev_rx_fillup` or `virtio_dev_status_update`... but I digress.